loopy/symbolic.py

```diff
@@ -1974,6 +1975,15 @@ def simplify_using_aff(kernel, expr):
             .get_inames_domain(inames)
             .project_out_except(inames, [dim_type.set]))
 
+    non_inames = deps - set(domain.get_var_dict().keys())
+    non_inames = set([name for name in set(non_inames) if name.isidentifier()])
+    if non_inames:
+        try:
+            extra_domain = isl.BasicSet("{[" + ",".join(non_inames) + "]}")
+            domain = domain.product(extra_domain)
+        except isl.Error:
+            pass
```

Comment:
Just call `domain.insert_dims` and then set the names.

Reply:
Done in 3a0d8c0